codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ConstrainedStringGeneratorTest.kt

```diff
@@ -184,4 +186,26 @@ class ConstrainedStringGeneratorTest {
 
         project.compileAndTest()
     }
+
+    @Test
+    fun `A regex that is accepted by Smithy but not by the regex crate cause tests to fail`() {
+        val model = """
+            namespace test
+
+            @pattern("import (?!static).+")
+            string PatternStringWithLookahead
+        """.asSmithyModel()
+
+        val constrainedStringShape = model.lookup<StringShape>("test#PatternStringWithLookahead")
+        val codegenContext = serverTestCodegenContext(model)
+        val project = TestWorkspace.testProject(codegenContext.symbolProvider)
+
+        project.withModule(ModelsModule) {
+            ConstrainedStringGenerator(codegenContext, this, constrainedStringShape).render()
+        }
+
+        assertThrows<CommandFailed> {
+            project.compileAndTest()
+        }
```

Comment:
Can we in any way verify the message it fails with, to make sure it fails for the reason we expect?

Reply:
yeah—you can look at the error message. An example: 
```
        val failure = shouldThrow<CommandFailed> { "cargo test".runWithWarnings(testDir) }
        failure.output shouldContain "endpoint::test::test_1"
        failure.output shouldContain "https://failingtest.com"
```

Reply:
Lovely, thanks!